test/ProtocalTests.cpp

```diff
@@ -0,0 +1,75 @@
+/*************************************************************************
+ * Copyright (c) 2023 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * See LICENSE.txt for license information
+ ************************************************************************/
+#include "TestBed.hpp"
+
+namespace RcclUnitTesting
+{
+  TEST(Protocol, Simple)
+  {
+    char env[] = "NCCL_PROTO=Simple";
+    putenv(env);
+
+    TestBed testBed;
+
+    // Configuration
+    std::vector<ncclFunc_t>     const funcTypes       = {ncclCollAllReduce};
+    std::vector<ncclDataType_t> const dataTypes       = {ncclFloat32};
+    std::vector<ncclRedOp_t>    const redOps          = {ncclSum};
+    std::vector<int>            const roots           = {0};
+    std::vector<int>            const numElements     = {1250};
+    std::vector<bool>           const inPlaceList     = {false};
+    std::vector<bool>           const managedMemList  = {false};
+    std::vector<bool>           const useHipGraphList = {false};
+
+    testBed.RunSimpleSweep(funcTypes, dataTypes, redOps, roots, numElements,
+                           inPlaceList, managedMemList, useHipGraphList);
```

Comment:
I would expect timeout requirement being passed in here?

Reply:
@wenkaidu do you think I should add it here as a param in the header, too?
Because I made it that user (or developer, through `putenv` in top level unit test code) can set timeout option in environment variable, so I made `RunSimpleSweep` just call timeout option from `ev` and pass to `ExecuteCollectives` in line 621.

Reply:
Normally CI will execute these tests. The timeout needs to have default value. Developer may override through env var if needed